x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/DateRangeBlockBuilder.java

```diff
@@ -0,0 +1,187 @@
+        return fromBuilder.estimatedBytes() + toBuilder.estimatedBytes();
+    }
+
+    @Override
+    public DateRangeBlockBuilder copyFrom(Block b, int beginInclusive, int endExclusive) {
+        Block fromBlock;
+        Block toBlock;
+        if (b.areAllValuesNull()) {
+            fromBlock = b;
+            toBlock = b;
+        } else {
+            var block = (DateRangeArrayBlock) b;
+            fromBlock = block.getFromBlock();
+            toBlock = block.getToBlock();
+        }
+        fromBuilder.copyFrom(fromBlock, beginInclusive, endExclusive);
+        toBuilder.copyFrom(toBlock, beginInclusive, endExclusive);
+        return this;
+    }
+
+    public DateRangeBlockBuilder copyFrom(DateRangeBlock block, int pos) {
+        if (block.isNull(pos)) {
+            appendNull();
+            return this;
+        }
+
+        if (block.getFromBlock().isNull(pos)) {
+            from().appendNull();
+        } else {
+            from().appendLong(block.getFromBlock().getLong(pos));
+        }
+
+        if (block.getToBlock().isNull(pos)) {
+            to().appendNull();
+        } else {
+            to().appendLong(block.getToBlock().getLong(pos));
+        }
+        return this;
+    }
+
+    @Override
+    public DateRangeBlockBuilder appendNull() {
+        fromBuilder.appendNull();
+        toBuilder.appendNull();
+        return this;
+    }
+
+    public DateRangeBlockBuilder appendDateRange(DateRangeLiteral lit) {
+        if (lit.from == null) {
+            fromBuilder.appendNull();
+        } else {
+            fromBuilder.appendLong(lit.from);
+        }
+        if (lit.to == null) {
+            toBuilder.appendNull();
+        } else {
+            toBuilder.appendLong(lit.to);
+        }
+        return this;
+    }
+
+    @Override
+    public DateRangeBlockBuilder mvOrdering(Block.MvOrdering mvOrdering) {
+        fromBuilder.mvOrdering(mvOrdering);
+        toBuilder.mvOrdering(mvOrdering);
+        return this;
+    }
+
+    @Override
+    public DateRangeBlock build() {
+        LongBlock fromBlock = null;
+        LongBlock toBlock = null;
+        boolean success = false;
+        try {
+            finish();
+            fromBlock = fromBuilder.build();
+            toBlock = toBuilder.build();
+            var block = new DateRangeArrayBlock(fromBlock, toBlock);
+            success = true;
+            return block;
+        } finally {
+            if (success == false) {
+                Releasables.closeExpectNoException(fromBlock, toBlock);
+            }
+        }
+    }
+
+    @Override
+    protected void extraClose() {
+        Releasables.closeExpectNoException(fromBuilder, toBuilder);
+    }
+
+    @Override
+    public BlockLoader.LongBuilder from() {
+        return fromBuilder;
+    }
+
+    @Override
+    public BlockLoader.LongBuilder to() {
+        return toBuilder;
+    }
+
+    public record DateRangeLiteral(Long from, Long to) implements GenericNamedWriteable {
```

Comment:
The word `Literal` in the class name makes me think this is generated by the ES|QL parser, but as far as I can tell it is not used for that at all. It is just a container for a single date-range value. Perhaps the class should be `DateRange` or `DateRangeValue`?